### ISSUE TITLE
createRoom fails with a 500 error if a badly-formatted mxid is in the `invitees` list

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#4088]
For example, if you pass an MXID which has a trailing space.

See https://github.com/matrix-org/synapse/issues/4032#issuecomment-429282870 for a stacktrace.

**Comment by RationalPhilomath** [comment]
Hey.. I would like to take this up. However, I have no prior experience working on any open source project before, and neither matrix.
Therefore, I would require support too.
Could I please take this up?

**Comment by f35f0ef9d0e827dae86552d3899f78fc** [comment]
Reading through linked conversations, looks like [parse_and_validate_server_name](https://github.com/matrix-org/synapse/blob/466866a1d9dd1fcf82348a36c0532cb0c6614767/synapse/http/endpoint.py#L49) function could be used.

**Comment by werner291** [comment]
Confirmed that this is still happening in latest Develop branch at time of writing (https://github.com/matrix-org/synapse/commit/54569c787b4abbc5674d9c23c012b56d8cc156ef).

Adding the following two tests in tests.rest.client.v1.test_rooms.RoomsCreateTestCase reproduces the issue:

```
    def test_post_inviteusers_mxid(self):
        # POST with list of invitee, expect new room id
        request, channel = self.make_request("POST", "/createRoom", b'{ "invite":["@alice:matrix.org"] }')

        self.render(request)
        self.assertEquals(200, channel.code, channel.result)
        self.assertTrue("room_id" in channel.json_body)

    def test_post_inviteusers_mxid_misformatted(self):
        # POST with invitee with trailing space in JSON, should return error 400 bad request.
        # Test created in relation to: https://github.com/matrix-org/synapse/issues/4088
        request, channel = self.make_request("POST", "/createRoom", b'{ "invite":["@alice:matrix.org "] }')

        self.render(request)
        self.assertEquals(400, channel.code, channel.result)
```

**Comment by werner291** [comment]
Appears to be an unhandled TypeError here: https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/message.py#L826

Something about `o.keys() are not iterable`. I'll try to figure out what's going on there.

**Comment by werner291** [comment]
returned_invite.signatures isn't iterable,apparently

**Comment by werner291** [comment]
Ok... Looks like there's just insufficient federation functionality being mocked to perform tests here.

**Comment by werner291** [comment]
:man_facepalming: Ok, I came to the brilliant conclusion that the FederationHandler during testing doesn't have a mock implementation of room invitation.

**Comment by werner291** [comment]
![image](https://user-images.githubusercontent.com/1017401/65732057-60957780-e0c8-11e9-986b-13dc356f322b.png)

Actually does still occur, though!

**Comment by werner291** [comment]
A stack trace:
```
Sep 26 23:43:40 nixos synapse[18978]: synapse.http.server: [POST-144] Failed handle request via 'RoomCreateRestServlet': <XForwardedForRequest at 0x7fc8444a8410 method='POST' uri='/_matrix/client/r0/createRoom' clientproto='HTTP/1.0' site=8008>
                                      Traceback (most recent call last):
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                      StopIteration: []
                                      
                                      During handling of the above exception, another exception occurred:
                                      
                                      Traceback (most recent call last):
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
                                          result = g.send(result)
                                      StopIteration: _RoutingResult(host_header=b'matrix.org ', tls_server_name=b'matrix.org ', target_host=b'matrix.org ', target_port=8448)
                                      
                                      During handling of the above exception, another exception occurred:
                                      
                                      Traceback (most recent call last):
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 424, in _send_request
                                          response = yield request_deferred
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
                                          result = g.send(result)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/federation/matrix_federation_agent.py", line 126, in request
                                          res.tls_server_name.decode("ascii")
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 132, in get_options
                                          return SSLClientConnectionCreator(host, ssl_context, should_verify)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 167, in __init__
                                          self._verifier = ConnectionVerifier(hostname, verify_certs)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 205, in __init__
                                          self._hostnameBytes = idna.encode(hostname)
                                        File "/nix/store/4ai7zwqa04ilzk0x4jxmqzqig39ikjjs-python3.7-idna-2.8/lib/python3.7/site-packages/idna/core.py", line 358, in encode
                                          s = alabel(label)
                                        File "/nix/store/4ai7zwqa04ilzk0x4jxmqzqig39ikjjs-python3.7-idna-2.8/lib/python3.7/site-packages/idna/core.py", line 270, in alabel
                                          ulabel(label)
                                        File "/nix/store/4ai7zwqa04ilzk0x4jxmqzqig39ikjjs-python3.7-idna-2.8/lib/python3.7/site-packages/idna/core.py", line 304, in ulabel
                                          check_label(label)
                                        File "/nix/store/4ai7zwqa04ilzk0x4jxmqzqig39ikjjs-python3.7-idna-2.8/lib/python3.7/site-packages/idna/core.py", line 261, in check_label
                                          raise InvalidCodepoint('Codepoint {0} at position {1} of {2} not allowed'.format(_unot(cp_value), pos+1, repr(label)))
                                      idna.core.InvalidCodepoint: Codepoint U+0020 at position 4 of 'org ' not allowed
                                      
                                      The above exception was the direct cause of the following exception:
                                      
                                      Traceback (most recent call last):
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/server.py", line 76, in wrapped_request_handler
                                          await h(self, request)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/server.py", line 315, in _async_render
                                          callback_return = await callback_return
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/rest/client/v1/room.py", line 91, in on_POST
                                          requester, self.get_room_config(request)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room.py", line 651, in create_room
                                          content=content,
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 302, in update_membership
                                          require_consent=require_consent,
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 508, in _update_membership
                                          require_consent=require_consent,
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 196, in _local_membership_update
                                          requester, event, context, extra_users=[target], ratelimit=ratelimit
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/util/metrics.py", line 69, in measured_func
                                          r = yield func(self, *args, **kwargs)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/message.py", line 705, in handle_new_client_event
                                          requester, event, context, ratelimit=ratelimit, extra_users=extra_users
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/message.py", line 783, in persist_and_notify_client_event
                                          invitee.domain, event
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/federation.py", line 1085, in send_invite
                                          pdu=event,
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/federation_client.py", line 742, in send_invite
                                          content = yield self._do_send_invite(destination, pdu, room_version)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/federation_client.py", line 783, in _do_send_invite
                                          "invite_room_state": pdu.unsigned.get("invite_room_state", []),
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/transport/client.py", line 314, in send_invite_v2
                                          destination=destination, path=path, data=content, ignore_backoff=True
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 640, in put_json
                                          timeout=timeout,
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 241, in _send_request_with_optional_trailing_slash
                                          response = yield self._send_request(request, **send_request_args)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
                                          result = result.throwExceptionIntoGenerator(g)
                                        File "/nix/store/kv604h4xyjdcvnfqxzprfr3msg9i1dvh-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
                                          return g.throw(self.type, self.value, self.tb)
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 429, in _send_request
                                          raise_from(RequestSendFailed(e, can_retry=True), e)
                                        File "<string>", line 3, in raise_from
                                      synapse.api.errors.RequestSendFailed: Failed to send request: InvalidCodepoint: Codepoint U+0020 at position 4 of 'org ' not allowed
```

**Comment by werner291** [comment]
Synapse-lines grepped out of it:
```
Sep 26 23:43:40 nixos synapse[18978]: synapse.http.server: [POST-144] Failed handle request via 'RoomCreateRestServlet': <XForwardedForRequest at 0x7fc8444a8410 method='POST' uri='/_matrix/client/r0/createRoom' clientproto='HTTP/1.0' site=8008>
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 424, in _send_request
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/federation/matrix_federation_agent.py", line 126, in request
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 132, in get_options
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 167, in __init__
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/crypto/context_factory.py", line 205, in __init__
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/server.py", line 76, in wrapped_request_handler
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/server.py", line 315, in _async_render
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/rest/client/v1/room.py", line 91, in on_POST
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room.py", line 651, in create_room
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 302, in update_membership
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 508, in _update_membership
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/room_member.py", line 196, in _local_membership_update
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/util/metrics.py", line 69, in measured_func
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/message.py", line 705, in handle_new_client_event
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/message.py", line 783, in persist_and_notify_client_event
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/handlers/federation.py", line 1085, in send_invite
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/federation_client.py", line 742, in send_invite
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/federation_client.py", line 783, in _do_send_invite
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/federation/transport/client.py", line 314, in send_invite_v2
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 640, in put_json
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 241, in _send_request_with_optional_trailing_slash
                                        File "/nix/store/02my60ip60g2vw2rhvynz79iihvs5v1i-matrix-synapse-1.3.1/lib/python3.7/site-packages/synapse/http/matrixfederationclient.py", line 429, in _send_request
                                      synapse.api.errors.RequestSendFailed: Failed to send request: InvalidCodepoint: Codepoint U+0020 at position 4 of 'org ' not allowed
```

**Comment by richvdh** [comment]
hopefully fixed by #6121
